### PR TITLE
Java 29169 :- Upgrade spring-jdbc to Spring Boot 3

### PR DESCRIPTION
**Pull request by amit2103**
The changes are :- 

1. Made Spring Boot 3 the parent for the Spring JDBC project.
2. Removed the deprecated methods in Spring JDBC.
3. Made change in the POM to avoid repackaging , Boot-2 parent POM didn't have repackaging enabled, 3 has.
4. Made some minor change to adhere to the post JDK 7 generics (using <> for initialization of objects) and removed an unused imports.

### CODE REVIEW

File: persistence-modules/spring-jdbc/src/main/java/com/baeldung/spring/jdbc/autogenkey/repository/MessageRepositoryJDBCTemplate.java [hunk 1]
```diff
@@ -21,7 +22,8 @@ public long insert(final String message) {
         KeyHolder keyHolder = new GeneratedKeyHolder();
 
         jdbcTemplate.update(connection -> {
```
Comment (anastasiosioannidis) [inline persistence-modules/spring-jdbc/src/main/java/com/baeldung/spring/jdbc/autogenkey/repository/MessageRepositoryJDBCTemplate.java:24]: Is Statement.RETURN_GENERATED_KEYS needed?

File: persistence-modules/spring-jdbc/src/main/java/com/baeldung/spring/jdbc/template/inclause/EmployeeDAO.java [hunk 2]
```diff
@@ -35,8 +35,7 @@ public List<Employee> getEmployeesFromIdList(List<Integer> ids) {
         String inSql = String.join(",", Collections.nCopies(ids.size(), "?"));
         List<Employee> employees = jdbcTemplate.query(
           String.format("SELECT * FROM EMPLOYEE WHERE id IN (%s)", inSql),
-          ids.toArray(),
-          (rs, rowNum) -> new Employee(rs.getInt("id"), rs.getString("first_name"), rs.getString("last_name")));
```
Comment (anastasiosioannidis) [inline persistence-modules/spring-jdbc/src/main/java/com/baeldung/spring/jdbc/template/inclause/EmployeeDAO.java:39]: Add a space between comma and "ids.toArray());"

File: persistence-modules/spring-jdbc/src/main/java/com/baeldung/spring/jdbc/replacedeprecated/StudentDaoWithDeprecatedJdbcTemplateMethods.java [hunk 3]
```diff
@@ -21,34 +21,28 @@ public void setJdbcTemplate(JdbcTemplate jdbcTemplate) {
 
     public List<Student> getStudentsOfAgeAndGender(Integer age, String gender) {
         String sql = "select student_id, student_name, age, gender, grade, state from student where age= ? and gender = ?";
-        Object[] args = {age, gender};
-        return jdbcTemplate.query(sql, args, new StudentRowMapper());
```
Comment (anastasiosioannidis) [inline persistence-modules/spring-jdbc/src/main/java/com/baeldung/spring/jdbc/replacedeprecated/StudentDaoWithDeprecatedJdbcTemplateMethods.java:25]: Add a space before "age"

File: persistence-modules/spring-jdbc/src/main/java/com/baeldung/spring/jdbc/autogenkey/repository/MessageRepositoryJDBCTemplate.java (hunk 1 above)
Comment (amit2103) [reply persistence-modules/spring-jdbc/src/main/java/com/baeldung/spring/jdbc/autogenkey/repository/MessageRepositoryJDBCTemplate.java:24]: Yes without it the test fails for me

File: persistence-modules/spring-jdbc/src/main/java/com/baeldung/spring/jdbc/replacedeprecated/StudentDaoWithDeprecatedJdbcTemplateMethods.java (hunk 3 above)
Comment (amit2103) [reply persistence-modules/spring-jdbc/src/main/java/com/baeldung/spring/jdbc/replacedeprecated/StudentDaoWithDeprecatedJdbcTemplateMethods.java:25]: Done

**Pull request merged**